tests/unit/test_sequence.py

```diff
@@ -174,3 +181,45 @@ def test_get_sequence_set():
     with pytest.raises(ValueError):
         seq_set = get_sequence_set(empty_lf)
         assert len(seq_set) == 0
+
+
+def test_filter_sequence_data(test_file_path, tmpdir):
+    test_sequence_file = test_file_path / "test_sequence.xz"
+    test_sequence_set = {
+        "USA/MD-MDH-1820/2021",
+        "USA/CA-CDPH-A3000000297958/2023",
+        "USA/WV064580/2020",
+        "USA/PA-CDC-LC1096774/2024",
+        "STARFLEET/DS9-DS9-001/2024",
+    }
+    mock_download = MagicMock(return_value=test_sequence_file, name="_download_from_url_mock")
+    with patch("cladetime.sequence._download_from_url", mock_download):
+        filtered_sequence_file, seq_total, seq_filtered = filter_sequence_data(
+            test_sequence_set, "http://thisismocked.com", tmpdir
+        )
+
+    assert filtered_sequence_file.exists()
```

Comment:
non-required comment: it would be extra reassuring to check that the filtered file's contents are correct!

Reply:
Just pushed an update to verify that the headers in the new .fasta are correct